multiapps-controller-core/src/main/java/org/cloudfoundry/multiapps/controller/core/cf/clients/CustomControllerClient.java

```diff
@@ -41,6 +57,14 @@ private PaginationV3 addPageOfResources(String uri, ResourcesResponseMapper<?> r
         return PaginationV3.fromResponse(responseMap);
     }
 
+    private MultiValueMap<String, String> generateRequestHeaders(String correlationId) {
+        Map<String, List<String>> map = headerConfiguration.generateHeaders(correlationId)
+                                                           .entrySet()
+                                                           .stream()
+                                                           .collect(Collectors.toMap(Map.Entry::getKey, e -> Arrays.asList(e.getValue())));
+        return CollectionUtils.toMultiValueMap(map);
```

Comment:
I would recommend doing this instead of using streams:
```java
var result = LinkedMultiValueMap<String, String>();
headerConfi.generateHeaders(correlationId).forEach(result::add);
return result;
```
The `add` method accepts a key and a single value and inserts it into a list value.
That way we can avoid some of the technicalities when converting from a normal to a multi-value map.